File: Source/OldCode/Archive/2004 TVA Code Library/Source/EE/Phasor/DataCellParsingState.vb
'*******************************************************************************************************
'  DataCellParsingState.vb - Data frame cell parsing state class
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2003
'  Primary Developer: James R Carroll, System Analyst [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  01/14/2005 - James R Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

Namespace EE.Phasor

    ' This class represents the protocol independent common implementation the parsing state of a data frame that can be sent or received from a PMU.
    Public Class DataCellParsingState

        Inherits ChannelCellParsingStateBase
        Implements IDataCellParsingState

        Private m_phasorValueType As Type
        Private m_frequencyValueType As Type
        Private m_analogValueType As Type
        Private m_digitalValueType As Type
        Private m_configurationCell As IConfigurationCell

        Public Sub New(ByVal phasorValueType As Type, ByVal frequencyValueType As Type, ByVal analogValueType As Type, ByVal digitalValueType As Type, ByVal configurationCell As IConfigurationCell)

            m_phasorValueType = phasorValueType
            m_frequencyValueType = frequencyValueType
            m_analogValueType = analogValueType
            m_digitalValueType = digitalValueType
            m_configurationCell = configurationCell

            With configurationCell
                PhasorCount = .PhasorDefinitions.Count
                AnalogCount = .AnalogDefinitions.Count
                DigitalCount = .DigitalDefinitions.Count
            End With

        End Sub

        Public Overrides ReadOnly Property InheritedType() As System.Type
            Get
                Return Me.GetType()
            End Get
        End Property

        Public ReadOnly Property ConfigurationCell() As IConfigurationCell Implements IDataCellParsingState.ConfigurationCell
            Get
                Return m_configurationCell
            End Get
        End Property

        Public ReadOnly Property PhasorValueType() As Type Implements IDataCellParsingState.PhasorValueType
            Get
                Return m_phasorValueType
            End Get
        End Property

        Public ReadOnly Property FrequencyValueType() As Type Implements IDataCellParsingState.FrequencyValueType
            Get
                Return m_frequencyValueType
            End Get
        End Property

        Public ReadOnly Property AnalogValueType() As Type Implements IDataCellParsingState.AnalogValueType
            Get
                Return m_analogValueType
            End Get
        End Property

        Public ReadOnly Property DigitalValueType() As Type Implements IDataCellParsingState.DigitalValueType
            Get
                Return m_digitalValueType
            End Get
        End Property

    End Class

End Namespace

File: Source/OldCode/Archive/2005 TVA Code Library/Version/4.0/Source/TVA.PhasorProtocols/DataCellParsingState.cs
//*******************************************************************************************************
//  DataCellParsingState.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  01/14/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace TVA.PhasorProtocols
{
    /// <summary>
    /// Represents the protocol independent common implementation of the parsing state used by any <see cref="IDataCell"/>.
    /// </summary>
    public class DataCellParsingState : ChannelCellParsingStateBase, IDataCellParsingState
    {
        #region [ Members ]

        // Fields
        private IConfigurationCell m_configurationCell;
        private CreateNewValueFunction<IPhasorDefinition, IPhasorValue> m_createNewPhasorValue;
        private CreateNewValueFunction<IFrequencyDefinition, IFrequencyValue> m_createNewFrequencyValue;
        private CreateNewValueFunction<IAnalogDefinition, IAnalogValue> m_createNewAnalogValue;
        private CreateNewValueFunction<IDigitalDefinition, IDigitalValue> m_createNewDigitalValue;

        #endregion

        #region [ Constructors ]

        /// <summary>
        /// Creates a new <see cref="DataCellParsingState"/> from specified parameters.
        /// </summary>
        /// <param name="configurationCell">Reference to the <see cref="IConfigurationCell"/> associated with the <see cref="IDataCell"/> being parsed.</param>
        /// <param name="createNewPhasorValue">Reference to delegate to create new <see cref="IPhasorValue"/> instances.</param>
        /// <param name="createNewFrequencyValue">Reference to delegate to create new <see cref="IFrequencyValue"/> instances.</param>
        /// <param name="createNewAnalogValue">Reference to delegate to create new <see cref="IAnalogValue"/> instances.</param>
        /// <param name="createNewDigitalValue">Reference to delegate to create new <see cref="IDigitalValue"/> instances.</param>
        public DataCellParsingState(IConfigurationCell configurationCell, CreateNewValueFunction<IPhasorDefinition, IPhasorValue> createNewPhasorValue, CreateNewValueFunction<IFrequencyDefinition, IFrequencyValue> createNewFrequencyValue, CreateNewValueFunction<IAnalogDefinition, IAnalogValue> createNewAnalogValue, CreateNewValueFunction<IDigitalDefinition, IDigitalValue> createNewDigitalValue)
        {
            m_configurationCell = configurationCell;
            m_createNewPhasorValue = createNewPhasorValue;
            m_createNewFrequencyValue = createNewFrequencyValue;
            m_createNewAnalogValue = createNewAnalogValue;
            m_createNewDigitalValue = createNewDigitalValue;

            PhasorCount = m_configurationCell.PhasorDefinitions.Count;
            AnalogCount = m_configurationCell.AnalogDefinitions.Count;
            DigitalCount = m_configurationCell.DigitalDefinitions.Count;
        }

        #endregion

        #region [ Properties ]

        /// <summary>
        /// Gets reference to the <see cref="IConfigurationCell"/> associated with the <see cref="IDataCell"/> being parsed.
        /// </summary>
        public virtual IConfigurationCell ConfigurationCell
        {
            get
            {
                return m_configurationCell;
            }
        }

        /// <summary>
        /// Gets reference to <see cref="CreateNewValueFunction{TDefinition,TValue}"/> delegate used to create new <see cref="IPhasorValue"/> objects.
        /// </summary>
        public virtual CreateNewValueFunction<IPhasorDefinition, IPhasorValue> CreateNewPhasorValue
        {
            get
            {
                return m_createNewPhasorValue;
            }
        }

        /// <summary>
        /// Gets reference to <see cref="CreateNewValueFunction{TDefinition,TValue}"/> delegate used to create new <see cref="IFrequencyValue"/> objects.
        /// </summary>
        public virtual CreateNewValueFunction<IFrequencyDefinition, IFrequencyValue> CreateNewFrequencyValue
        {
            get
            {
                return m_createNewFrequencyValue;
            }
        }

        /// <summary>
        /// Gets reference to <see cref="CreateNewValueFunction{TDefinition,TValue}"/> delegate used to create new <see cref="IAnalogValue"/> objects.
        /// </summary>
        public virtual CreateNewValueFunction<IAnalogDefinition, IAnalogValue> CreateNewAnalogValue
        {
            get
            {
                return m_createNewAnalogValue;
            }
        }

        /// <summary>
        /// Gets reference to <see cref="CreateNewValueFunction{TDefinition,TValue}"/> delegate used to create new <see cref="IDigitalValue"/> objects.
        /// </summary>
        public virtual CreateNewValueFunction<IDigitalDefinition, IDigitalValue> CreateNewDigitalValue
        {
            get
            {
                return m_createNewDigitalValue;
            }
        }

        #endregion
    }
}